actions_test.go

```diff
@@ -73,13 +116,51 @@ func TestAction_RemoveMatcher(t *testing.T) {
 func TestAction_AddPath(t *testing.T) {
 	t.Parallel()
 
+	const envGitHubPath = "GITHUB_PATH"
```

Comment:
It's generally a smell to set/unset envvars in tests, as it causes races when run in parallel. Can you tell me more about what this is attempting to do? I might be able to suggest an alternative approach.

Reply:
So essentially with new file commands, GitHub actions populate the environment with `GITHUB_{CMD}` variables, e.g. `GITHUB_PATH` with a file path where command data needs to be written. This is basically to test our code's behaviour end-to-end. We pass the same environment variable with a file path to it, that we expected from the GitHub runner. I originally didn't unset the env variables so when the tests ran in GitHub action environment, they failed because first part - of the tests - tests the fallback commands in absence of the variable. The issue obviously was that these variables were being populated by the runner.

Reply:
Could we just set `GITHUB_PATH=""` inside the action.yml instead?

Reply:
I think we can. Let me try!

Reply:
@sethvargo We're still stuck with 1 Setenv call per test for three test cases. The reason being that `IssueFileCommand` expects the file path to be passed as an environment variable, and `AddPath` and `SetEnv` are both dependent on it. Although resetting required env vars in workflow config does remove the reset/restore calls in tests, they still need to set for completing the tests.

Could you go through [`AddPath`](https://github.com/sethvargo/go-githubactions/blob/a03a58270e57ec037ff19873ab13a086ce527b65/actions_test.go#L116-L164) test again?

https://github.com/sethvargo/go-githubactions/blob/a03a58270e57ec037ff19873ab13a086ce527b65/actions_test.go#L116-L164

In words, it does the following:

1. Unset `GITHUB_PATH` (provided by the GitHub runner)
2. Call `AddPath()` since `GITHUB_PATH` is unset, the test expects the fallback command to execute
3. Create a temp file
4. Set `GITHUB_PATH` to the temp file path
5. Call `AddPath()` again and expect the file command to execute
6. Verify if command data was written to the temp file
7. File cleanup and env restoration

`SetEnv` test also follows the same pattern. Now even if we eliminate the `Unset` call, we still have to populate `GITHUB_PATH` with the temp file path.

Reply:
I originally thought of that. Lacked the confidence to carry it out in effect. lol

Reply:
I thought of that at some point. Lacked the confidence to carry it out in effect. lol